commons/src/store/files.ts

```diff
@@ -0,0 +1,52 @@
+import { writable } from "svelte/store";
+import type { File, Message } from "@commons/types/Nylas";
+import { downloadFile } from "@commons/connections/files";
+
+function initializeFilesForMessage() {
+  const { subscribe, set, update } = writable<
+    Record<string, Record<string, File>>
+  >({});
+  const filesMap: Record<string, Record<string, File>> = {}; // {message_id: {file_id: Base64 String}}
+
+  return {
+    subscribe,
+    getFilesForMessage: async (
+      incomingMessage: Message,
+      query: { component_id: string; access_token: string },
+    ) => {
+      if (!filesMap[incomingMessage.id]) {
+        const inlineFiles: Record<string, File> = {};
+        for (const file of incomingMessage.files.values()) {
+          if (file.content_disposition === "inline") {
```

Comment:
Let's merge this with the condition below
```suggestion
          if (file.content_disposition === "inline" && !inlineFiles[file.id]) {
```